File: mobile/lib/models/order.dart
class OrderItem {
  final String artwork;
  final String title;
  final double ngnPrice;
  final double usdPrice;
  final int quantity;
  final String? image;

  OrderItem({
    required this.artwork,
    required this.title,
    required this.ngnPrice,
    required this.usdPrice,
    this.quantity = 1,
    this.image,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      artwork: json['artwork'] ?? '',
      title: json['title'] ?? '',
      ngnPrice: (json['price']?['ngn'] ?? 0).toDouble(),
      usdPrice: (json['price']?['usd'] ?? 0).toDouble(),
      quantity: json['quantity'] ?? 1,
      image: json['image'],
    );
  }
}

class Order {
  final String id;
  final String buyerId;
  final String artistId;
  final List<OrderItem> items;
  final double totalNgn;
  final double totalUsd;
  final String status;
  final String paymentStatus;
  final DateTime createdAt;

  Order({
    required this.id,
    required this.buyerId,
    required this.artistId,
    required this.items,
    required this.totalNgn,
    required this.totalUsd,
    this.status = 'Pending',
    this.paymentStatus = 'Unpaid',
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['_id'] ?? '',
      buyerId: json['buyer'] ?? '',
      artistId: json['artist'] ?? '',
      items: (json['items'] as List<dynamic>?)
              ?.map((e) => OrderItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      totalNgn: (json['totalAmount']?['ngn'] ?? 0).toDouble(),
      totalUsd: (json['totalAmount']?['usd'] ?? 0).toDouble(),
      status: json['status'] ?? 'Pending',
      paymentStatus: json['paymentStatus'] ?? 'Unpaid',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : null,
    );
  }
}

File: backend/models/Order.js
const mongoose = require('mongoose');

const orderItemSchema = new mongoose.Schema(
  {
    artwork: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Artwork',
      required: true,
    },
    title: String,
    price: {
      ngn: Number,
      usd: Number,
    },
    quantity: { type: Number, default: 1 },
    image: String,
  },
  { _id: false }
);

const orderSchema = new mongoose.Schema(
  {
    buyer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    artist: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    items: [orderItemSchema],
    totalAmount: {
      ngn: { type: Number, required: true },
      usd: { type: Number, required: true },
    },
    currency: {
      type: String,
      enum: ['NGN', 'USD'],
      default: 'NGN',
    },
    status: {
      type: String,
      enum: ['Pending', 'Confirmed', 'Processing', 'Shipped', 'Delivered', 'Cancelled'],
      default: 'Pending',
    },
    paymentStatus: {
      type: String,
      enum: ['Unpaid', 'Paid', 'Refunded', 'Failed'],
      default: 'Unpaid',
    },
    paymentReference: { type: String },
    paymentMethod: { type: String, default: 'Paystack' },
    shippingAddress: {
      fullName: { type: String, required: true },
      phone: { type: String, required: true },
      address: { type: String, required: true },
      city: { type: String, required: true },
      state: { type: String, required: true },
      country: { type: String, required: true },
      zipCode: { type: String },
    },
    deliveryNotes: { type: String, maxlength: 500 },
    trackingNumber: { type: String },
    estimatedDelivery: { type: Date },
    deliveredAt: { type: Date },
  },
  { timestamps: true }
);

orderSchema.index({ buyer: 1, createdAt: -1 });
orderSchema.index({ artist: 1, createdAt: -1 });
orderSchema.index({ status: 1 });

module.exports = mongoose.model('Order', orderSchema);
